alibi/explainers/similarity/grad.py

```diff
@@ -129,21 +129,22 @@ def __init__(self,
             task_name=task
         )
 
-        non_trainable_layers = self.backend.get_non_trainable(self.predictor)
+        non_trainable_layers = self.backend._get_non_trainable(self.predictor)
         if non_trainable_layers:
-            layers_msg = 'The following layers are not trainable: '
+            layers_msg = 'The following tensors are not trainable: '
             layers = ", ".join([f"'{layer}'" for layer in non_trainable_layers if layer is not None])
-            warning_msg = ("Some layers in the model are not trainable. These layer gradients will not be "
-                           f"included in the computation of gradient similarity. {layers_msg}{layers}")
-            warnings.warn(warning_msg)
+            warning_msg = ("Some layers in the model are not trainable. These layers don't have gradients "
+                           "and will not be included in the computation of gradient similarity. "
+                           f"{layers_msg}{layers}")
+            warnings.warn(warning_msg)  # todo: scope warning to this location
 
     def fit(self,
             X_train: np.ndarray,
             Y_train: np.ndarray) -> "Explainer":
         """Fit the explainer.
 
-        The GradientSimilarity explainer requires the model gradients over the training data. In the explain method it
-        compares them to the model gradients for the test instance(s). If ``store_grads`` was set to ``True`` on
+        The ``GradientSimilarity`` explainer requires the model gradients over the training data. In the explain method
+        it compares them to the model gradients for the test instance(s). If ``store_grads`` was set to ``True`` on
```

Comment:
Just noticed that the public parameter name is `precompute_grads` not `store_grads`.